// @p5-wrapper/react/dist/component/main.es.js
import * as p from "react";
const A = { Date: !0, RegExp: !0, String: !0, Number: !0 };
function f(r, s, c = { cyclesFix: !0 }, o = []) {
  let n = [];
  const l = Array.isArray(r);
  for (const t in r) {
    const e = r[t], i = l ? +t : t;
    if (!(t in s)) {
      n.push({
        type: "REMOVE",
        path: [i],
        oldValue: r[t]
      });
      continue;
    }
    const a = s[t], u = typeof e == "object" && typeof a == "object" && Array.isArray(e) === Array.isArray(a);
    e && a && u && !A[Object.getPrototypeOf(e)?.constructor?.name] && (!c.cyclesFix || !o.includes(e)) ? n.push.apply(n, f(e, a, c, c.cyclesFix ? o.concat([e]) : []).map((y) => (y.path.unshift(i), y))) : e !== a && // treat NaN values as equivalent
    !(Number.isNaN(e) && Number.isNaN(a)) && !(u && (isNaN(e) ? e + "" == a + "" : +e == +a)) && n.push({
      path: [i],
      type: "CHANGE",
      value: a,
      oldValue: e
    });
  }
  const h = Array.isArray(s);
  for (const t in s)
    t in r || n.push({
      type: "CREATE",
      path: [h ? +t : t],
      value: s[t]
    });
  return n;
}
function m(r, s) {
  return f(r, s).length === 0;
}
const N = p.lazy(() => import("./P5CanvasGuard-TAOYOyUP.js").then((r) => r.P)), C = p.memo(N, m), d = "canvas-container";
export {
  d as CanvasContainerClassName,
  C as P5Canvas
};
