// @p5-wrapper/react/dist/component/P5CanvasGuard-TAOYOyUP.js
import { jsx as c } from "react/jsx-runtime";
import * as d from "react";
import _ from "react";
var u = { exports: {} }, f = {};
var h;
function R() {
  if (h) return f;
  h = 1;
  var e = _.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE;
  return f.c = function(t) {
    return e.H.useMemoCache(t);
  }, f;
}
var m = {};
var g;
function v() {
  return g || (g = 1, process.env.NODE_ENV !== "production" && (function() {
    var e = _.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE;
    m.c = function(t) {
      var n = e.H;
      return n === null && console.error(
        `Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.`
      ), n.useMemoCache(t);
    };
  })()), m;
}
var p;
function y() {
  return p || (p = 1, process.env.NODE_ENV === "production" ? u.exports = R() : u.exports = v()), u.exports;
}
var S = y();
const E = "[P5Canvas] The error boundary was triggered. The error message was:";
function b(e) {
  return e.trim().split(`
`).map((t) => t.trim()).join(`
`);
}
function N(e) {
  return b(`
      ${E}
      
      ${e}
    `);
}
function O(e) {
  return e instanceof Error ? `${e.name}("${e.message}")` : typeof e == "symbol" || e instanceof Symbol ? e.toString() : typeof e == "string" || e instanceof String ? `String("${e.toString()}")` : typeof e == "number" || e instanceof Number ? `Number(${e.toString()})` : typeof e == "bigint" || e instanceof BigInt ? `BigInt(${e.toString()})` : e instanceof Array ? `Array(${JSON.stringify(e.values().toArray())})` : e instanceof Set ? `Set(${JSON.stringify(e.values().toArray())})` : Object.getPrototypeOf(e) === Object.prototype ? `Object(${JSON.stringify(e)})` : `Unknown(${typeof e})`;
}
function C(e) {
  const t = O(e), n = N(t);
  console.error(n);
}
const T = d.lazy(() => import("./P5CanvasWithSketch-XYe9fxPc.js")), k = d.lazy(() => import("./react-error-boundary-yKHK87iG.js").then((e) => ({
  default: e.ErrorBoundary
}))), $ = (e) => {
  const t = S.c(14);
  if (e.sketch === void 0) {
    console.error("[P5Canvas] The `sketch` prop is required.");
    let a;
    return t[0] !== e.fallback ? (a = e.fallback?.() ?? null, t[0] = e.fallback, t[1] = a) : a = t[1], a;
  }
  let n;
  t[2] !== e ? (n = (a) => e.error?.(a.error) ?? /* @__PURE__ */ c("p", { "data-testid": "error", children: "❌ - Something went wrong" }), t[2] = e, t[3] = n) : n = t[3];
  let o;
  t[4] !== e.loading ? (o = e.loading?.() ?? /* @__PURE__ */ c("p", { "data-testid": "loading", children: "🚀 Loading..." }), t[4] = e.loading, t[5] = o) : o = t[5];
  const l = e;
  let r;
  t[6] !== l ? (r = /* @__PURE__ */ c(T, { ...l }), t[6] = l, t[7] = r) : r = t[7];
  let i;
  t[8] !== o || t[9] !== r ? (i = /* @__PURE__ */ c(d.Suspense, { fallback: o, children: r }), t[8] = o, t[9] = r, t[10] = i) : i = t[10];
  let s;
  return t[11] !== n || t[12] !== i ? (s = /* @__PURE__ */ c(k, { fallbackRender: n, onError: A, children: i }), t[11] = n, t[12] = i, t[13] = s) : s = t[13], s;
};
function A(e) {
  C(e);
}
const P = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: $
}, Symbol.toStringTag, { value: "Module" }));
export {
  P,
  S as c
};
